backend/models/user.mgmodel.ts

```diff
@@ -10,24 +10,46 @@ export interface User extends Document {
   role: Role;
 }
 
-const UserSchema: Schema = new Schema({
-  firstName: {
-    type: String,
-    required: true,
-  },
-  lastName: {
-    type: String,
-    required: true,
-  },
-  authId: {
-    type: String,
-    required: true,
-  },
-  role: {
-    type: String,
-    required: true,
-    enum: ["Administrator", "Facilitator", "Learner"],
+const baseOptions = {
+  discriminatorKey: "type",
+  collection: "user",
```

Comment:
I think we can remove this `collection` option.

My understanding is that `collection` specifies the name of the collection for the model? You'll notice we already had a collection `users` and that specifying this option spawns another collection `user`. No need for both, just `users` (the existing one) is fine.

<img width="241" alt="Screenshot 2024-07-12 at 2 07 08 AM" src="https://github.com/user-attachments/assets/8341869b-787a-4c10-bbeb-7c52af8ccc66">


FYI - I believe the default behaviour is that Mongoose comes up with a collection name by taking the lowercase plural of the first argument passed to `mongoose.model`, e.g. `mongoose.model<User>("User", UserSchema)` gives you `users`.